versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -1132,23 +1176,44 @@ protected <T> void writeMany(
       String sqlInsert,
       List<T> entries,
       Function<T, String> idRetriever,
-      Function<T, byte[]> serializer)
-      throws ReferenceConflictException {
+      Function<T, byte[]> serializer,
+      boolean update)
+      throws ReferenceConflictException, ReferenceNotFoundException {
     int cnt = 0;
     try (PreparedStatement ps = c.conn().prepareStatement(sqlInsert)) {
       for (T e : entries) {
-        ps.setString(1, config.getRepositoryId());
-        ps.setString(2, idRetriever.apply(e));
-        ps.setBytes(3, serializer.apply(e));
+        if (update) {
+          ps.setBytes(1, serializer.apply(e));
+          ps.setString(2, config.getRepositoryId());
+          ps.setString(3, idRetriever.apply(e));
+        } else {
+          ps.setString(1, config.getRepositoryId());
+          ps.setString(2, idRetriever.apply(e));
+          ps.setBytes(3, serializer.apply(e));
+        }
         ps.addBatch();
         cnt++;
         if (cnt == config.getBatchSize()) {
-          ps.executeBatch();
+          int[] result = ps.executeBatch();
+          if (update) {
+            for (int i : result) {
+              if (i != 1) {
```

Comment:
nit: according to the docs this can also return `java.sql.Statement#SUCCESS_NO_INFO`, but i guess we can assume all the driver+db combinations provide this info here